### DISCUSSION TITLE
Add a JSON endpoint to get localization messages for a locale

### DISCUSSION THREAD

**Discussion by stelewis-redhat** [keycloak/keycloak#15841] [Feedback]
Currently, localized messages are added to the FreeMarker template so that the front end can have access to them.

If we are implementing React-based front-ends, it would be nice to access an endpoint to request a particular locale's messages. Something like:

/localizations/<locale>

I have a preliminary implementation that I've done (https://github.com/stelewis-redhat/keycloak/blob/accountformservice-localizations/services/src/main/java/org/keycloak/services/resources/account/AccountFormService.java#L341), but I have questions I'd like to raise:

1. Should it get GET or POST? I need to look into other examples of JSON endpoints and follow that pattern.
2. What about error handling? What do we want to do if there's an error, from invalid input, etc. I can look at other examples in the codebase.
3. What are the security implications around this? How secure does it need to be, if we are accessing this endpoint from JavaScript before the user even logs in?
4. I assume that, even though keycloak UI is a separate project, for the backend portions we'll still want to do a PR against keycloak itself.

**Comment by stelewis-redhat** [comment]
@jonkoops I went ahead and started a discussion.

**Comment by jonkoops** [comment]
> Should it get GET or POST? I need to look into other examples of JSON endpoints and follow that pattern.

Since this is retrieving a JSON file I think GET would be most appropriate here.

> What about error handling? What do we want to do if there's an error, from invalid input, etc. I can look at other examples in the codebase.

Good question, I am not too familiar with how this is done currently in other parts of the API.

>  What are the security implications around this? 

I'd argue this could be a public API that requires no authentication, but opinions might differ.

> I assume that, even though keycloak UI is a separate project, for the backend portions we'll still want to do a PR against keycloak itself.

Yes, we are however in the process of migrating the UIs to the main project. Although it will take a while before we are there.

**Reply by stelewis-redhat** [reply]
Thanks @jonkoops I'll look at how the admin endpoints are done around error handling.

**Reply by stelewis-redhat** [reply]
I'll be using this as reference: https://github.com/keycloak/keycloak-community/blob/main/design/rest-api-guideline.md. I thought I'd seen it before but found it again.
